docs/static/upgrading.asciidoc

```diff
@@ -77,3 +77,20 @@ of workers by passing a command line flag such as:
 
 [source,shell]
 bin/logstash `-w 1`
+
+=== Upgrading Logstash to 2.2
+
+Logstash 2.2 re-architected the pipeline stages to provide more performance and help future enhancements in resiliency. 
+The new pipeline introduced micro-batches, processing groups of events at a time. The default batch size is 
+125 per worker. Also, the filter and output stages are executed in the same thread, but still, as different stages.
+The CLI flag `--pipeline-workers` or `-w` control the number of execution threads, which is set by default to number of cores.
+
+**Performance Tuning in 2.2**
+Since both filters and output workers are on the same thread, this could lead to threads being idle in I/O wait state. 
+Thus, in 2.2, you can safely set `-w` to a number which is 2x number of cores. A common way to tune performance is 
+keep increasing the `-w` beyond the # of cores until performance no longer improves. A note of caution - make sure you also 
+keep heapsize in mind, because the number of in-flight events are `# of workers * batch_size`. More in-flight events 
```

Comment:
Might be worth completing the equation to be `#workers * batch_size * average_event size`